processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerModel.java

```diff
@@ -259,14 +259,16 @@ public static CarbonFactDataHandlerModel createCarbonFactDataHandlerModel(
     carbonFactDataHandlerModel.sortScope = CarbonDataProcessorUtil.getSortScope(configuration);
 
     DataMapWriterListener listener = new DataMapWriterListener();
-    listener.registerAllWriter(
-        configuration.getTableSpec().getCarbonTable(),
-        configuration.getSegmentId(),
-        CarbonTablePath.getShardName(
-            carbonDataFileAttributes.getTaskId(),
-            bucketId,
-            0,
-            String.valueOf(carbonDataFileAttributes.getFactTimeStamp())));
+    if (null != configuration.getSegmentId()) {
+      listener.registerAllWriter(
```

Comment:
add a comment for this. SDK currently does not support writing datamap

Reply:
ok, done